stores/filestore.go

```diff
@@ -665,6 +715,286 @@ func (w *bufferedWriter) checkShrinkRequest() {
 }
 
 ////////////////////////////////////////////////////////////////////////////
+// filesManager methods
+////////////////////////////////////////////////////////////////////////////
+
+// createFilesManager returns an instance of the files manager.
+func createFilesManager(rootDir string, openedFilesLimit int64) *filesManager {
+	fm := &filesManager{
+		rootDir: rootDir,
+		limit:   openedFilesLimit,
+		files:   make(map[fileID]*file),
+	}
+	return fm
+}
+
+// closeUnusedFiles cloes files that are opened and not currently in-use.
+// Since the number of opened files is a soft limit, and if this function
+// is unable to close any file, the caller will still attempt to create/open
+// the requested file. If the system's file descriptor limit is reached,
+// opening the file will fail and that error will be returned to the caller.
+// Lock is required on entry.
+func (fm *filesManager) _closeUnusedFiles(idToSkip fileID) {
```

Comment:
Yes, you got the meaning. But it is not a new convention that I would impose on the group. I would also welcome a better way to do this (without making `filesManager` public).